### PR TITLE
Add sourcemaps to all distribution of every package

### LINKED ISSUE #2785: Sourcemaps are missing from distribution packages and local builds
I'm trying to debug #2764 but finding it difficult to navigate the codebase with the Node.js Inspector because there are no source maps. Without sourcemaps, it's very difficult to place breakpoints by looking at TypeScript source code. This problem effects both contributors to Backstage and developers consuming Backstage.

## Expected Behavior

Backstage distribution packages should include working sourcemaps. Building a development environment locally should generate sourcemaps for local debugging.

## Current Behavior

Sourcemaps are not included in distribution packages. Sourcemaps are not generated when running the project locally from source.

## Possible Solution

For distribution packages, we need to configure `esbuild` to include sourcemaps. We might also need to include the TypeScript source code to ensure that sourcemaps have actual source that they reference. 

For local development, we need to generate sourcemaps and point them at `src`directories.  

## Steps to Reproduce

For distribution packages,

1. download a package from NPM and look at it's content
2. it'll be missing .map files

For local development,

1. Build project locally
2. dist will be missing .map files

### PR DESCRIPTION
**Pull request by taras**
## Hey, I just made a Pull Request!

Fixes #2785

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Review by stefanalund — APPROVED** [review]
@Rugvip this is your domain :)

File: .changeset/loud-lamps-visit.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/cli': minor
```
Comment (Rugvip) [inline .changeset/loud-lamps-visit.md:2]: Should be patch, no? 😅

**Review by Rugvip — APPROVED** [review]
Thanks! Source maps look good. 👍  Wanted to verify whether they get picked up properly in the bundling step too, but figured that'll be much lower effort to test once this is published 😅

File: .changeset/loud-lamps-visit.md (hunk 1 above)
Comment (taras) [reply .changeset/loud-lamps-visit.md]: Done in https://github.com/spotify/backstage/pull/2787/commits/2896a6f02d8033b84780612e6513598320caa4fc

**Review by freben — APPROVED** [review]

**Pull request merged**